crates/cadenza-syntax/src/parse.rs

```diff
@@ -573,9 +573,10 @@ impl<const CLOSE: u16> Marker for DelimiterMarker<CLOSE> {
         if current == Self::close_kind() || current == Kind::Eof {
             return false;
         }
-        // Delegate to saved whitespace marker for indentation checking
-        // It already handles commas specially (as it handles infix operators)
-        self.saved_whitespace.should_continue(parser)
+        
+        // Inside delimiters, always continue unless we're at the closing delimiter
+        // This allows both same-line and multi-line content without strict indentation rules
+        true
```

Comment:
This is what hurts recovery. This will end up consuming the rest of the entire file, leading to worse errors.